src/pages/Stream/Views/Manage/Info.tsx

```diff
@@ -13,35 +17,221 @@ const Header = () => {
-				{props.title}
-			</Text>
-			<Text
-				className={classes.fieldTitle}
-				style={{ textOverflow: 'ellipsis', whiteSpace: 'nowrap', overflow: 'hidden', fontWeight: 400 }}>
-				{props.value}
+				Max Historical Differernce
 			</Text>
+			<Stack
+				gap={2}
+				style={{
+					flexDirection: 'row',
+					alignItems: 'center',
+				}}>
+				<TextInput w={'100%'} placeholder="Max Historical Difference" value={value} onChange={(e) => onChange(e)} />
+				<UpdateFieldButtons onClick={onClick} onClose={props.onClose} updating={updating} />
+			</Stack>
+		</Stack>
+	);
+};
+
+const UpdateCustomPartitionField = (props: { isStaticSchema: boolean; onClose: () => void; currentStream: string }) => {
+	const [info] = useStreamStore((store) => store.info);
+	const updateableCustomPartition = _.get(info, 'custom_partition', '');
+	const partitionFields: string[] = updateableCustomPartition.split(',');
+	const [value, setValue] = useState<string[]>(partitionFields);
+	const [updating, setUpdating] = useState<boolean>(false);
+	const { updateLogStreamMutation } = useLogStream();
+	const { getStreamInfoRefetch } = useGetStreamInfo(props.currentStream);
+
+	const onChangeValue = useCallback(
+		(value: string[]) => {
+			setValue(value);
+		},
+		[setValue],
+	);
+
+	const updateLogStreamSuccess = useCallback(() => {
+		props.onClose();
+		setUpdating(false);
+		getStreamInfoRefetch();
+	}, [getStreamInfoRefetch]);
+
+	const updateLogStream = useCallback(
+		(updatedValue: string) => {
+			updateLogStreamMutation({
+				streamName: props.currentStream,
+				customPartition: updatedValue,
+				onSuccess: () => updateLogStreamSuccess(),
+				onError: () => setUpdating(false),
+			});
+		},
+		[props.currentStream, updateLogStreamMutation],
+	);
+
+	const onClick = useCallback(() => {
+		if (value === undefined) return;
+
+		const valuesFlattened = value.join(',');
+		setUpdating(true);
+		updateLogStream(valuesFlattened);
+	}, [value, updateLogStream]);
+
+	return (
+		<Stack gap={0} style={{ justifyContent: 'space-between' }}>
+			<Stack>
+				<Text className={classes.fieldDescription}>Custom Partition Field</Text>
+			</Stack>
+
+			<Stack style={{ flexDirection: 'row', alignItems: 'center' }}>
+				<TagsInput
+					w={'30rem'}
+					placeholder="Add upto 3 columns"
+					data={partitionFields}
+					onChange={(value) => onChangeValue(value)}
+					maxTags={3}
+					value={value}
+				/>
+				<UpdateFieldButtons onClick={onClick} onClose={props.onClose} updating={updating} />
+			</Stack>
 		</Stack>
 	);
 };
 
-const InfoData = (props: {isLoading: boolean}) => {
+const InfoItem = (props: { title: string; value: string; fullWidth?: boolean; editable?: boolean }) => {
+	const [showEditField, setShowEditField] = useState<boolean>(false);
+	const [currentStream] = useAppStore((store) => store.currentStream);
+	const closeEditField = useCallback(() => {
+		setShowEditField(false);
+	}, []);
+
+	return (
+		<>
```

Comment:
Again. This is not readable. 
Do not wrap components with <></> or React.Fragment if the content exceeds 5 to 10 lines. 
Extract them into separate components.